galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/common/GalasaProperty.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.api.cps.internal.common;
+
+import java.util.Map;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+import dev.galasa.framework.spi.utils.GalasaGsonBuilder;
+
+public class GalasaProperty {
+    static final Gson gson = GalasaGsonBuilder.build();
+    
+    public String apiVersion = "galasa-dev/v1alpha1";
+    public final String kind = "GalasaProperty"; 
+    public GalasaPropertyMetadata metadata ;
+    public GalasaPropertyData data;
+    
+    public class GalasaPropertyMetadata {
+        public String namespace;
+        public String name;
+        
+        public GalasaPropertyMetadata (String namespace, String name){
+            this.namespace = namespace;
+            this.name = name;
+        }
+    }
+
+    public class GalasaPropertyData {
+        public String value;
+        
+        public GalasaPropertyData (String propertyValue){
+            this.value = propertyValue;
+        }
+    }
+
+    public GalasaProperty (String completeCPSname, String propertyValue){
+        String[] name = completeCPSname.split("[.]", 2);
+        this.metadata = new GalasaPropertyMetadata(name[0],name[1]);
+        this.data = new GalasaPropertyData(propertyValue);
```

Comment:
Are we sure that the completeCPSname will split into two parts here? (or could we get an out-of-bounds error)